op-deployer/book/src/reference-guide/custom-superchains.md

```diff
@@ -0,0 +1,157 @@
+
+- Deploy contract implementations that will be shared among all OP Chains deployed using this Superchain.
+- Deploy Superchain-wide management contracts like `SuperchainConfig` and `SuperchainVersions`.
+- Set up ownership so that you can control the Superchain.
+
+You will use the `bootstrap` [family][bootstrap] of commands on `op-deployer` to do this.
+
+[bootstrap]: ../user-guide/bootstrap.md
+
+### Superchain Bootstrap
+
+To begin, bootstrap the Superchain onto your chosen L1 with the following command:
+
+```shell
+op-deployer bootstrap superchain \
+  --l1-rpc-url="<rpc url>" \
+  --private-key="<contract deployer private key>" \
+  --artifacts-locator="<locator>" \
+  --outfile="<path to outfile>" \
+  --superchain-proxy-admin-owner="<role address>" \
+  --protocol-versions-owner="<role address>" \
+  --guardian="<role address>"
+```
+
+This will output a JSON file containing the addresses of the relevant contract. Keep track of this file, as you will
+need it in subsequent steps.
+
+We recommend the following these best practices when bootstrapping the Superchain:
+
+1. Use Gnosis SAFEs for ownership roles like `guardian` and `superchain-proxy-admin-owner`. The owner **must** be a
+   smart contract to support future upgrades, so a SAFE is a sensible default.
+2. Use a regular EOA as the deployer. It will not have any control over the Superchain once the deployment completes.
+3. Use a standard contracts tag (e.g. `tag://op-contracts/v2.0.0`). This will make upgrading easier.
+
+### Bootstrapping Implementations
+
+Next, you will need to bootstrap implementations onto the L1. This is done with the following command:
+
+```shell
+op-deployer bootstrap implementations \
+  --artifacts-locator="<locator, should be the same as the one used in bootstrap superchain>" \
+  --l1-rpc-url="<rpc url>" \
+  --outfile="<path to outfile>" \
+  --mips-version="2" \
+  --private-key="<contract deployer private key>" \
+  --protocol-versions-proxy="<address output from bootstrap superchain>" \
+  --superchain-config-proxy="<address output from bootstrap superchain>" \
+  --upgrade-controller="<superchain-proxy-admin-owner used in bootstrap superchain>"
+```
+
+Similar to the Superchain bootstrap command, this will output a JSON file containing the addresses of the relevant
+contracts. Again, keep track of this file. The deployment scripts use `CREATE2` under the hood, so it will only
+deploy contracts if their constructor arguments or implementations change. This will save time and gas.
+
+The most important address in the implementations file is the OPCM, or OP Contracts Manager. This contract is the
+factory that will deploy all the OP Chains belonging to this Superchain. It is also responsible for upgrading between
+different contracts versions. There is a one-to-one mapping between OPCM, and contracts version. For this reason, it
+is very important to use standard contracts versions in your deployment.
+
```

Comment:
```suggestion

> [!warning]
> If you are deploying a custom L2 that points to Ethereum Sepolia or Ethereum Mainnet (e.g., NOT an L3), you do not need to bootstrap a new L1, since Ethereum Sepolia/Mainnet already have the necessary contracts. However, you _will_ need to deploy a new OPCM through `op-deployer bootstrap implementations`.
```

@mslipper -- is this right? this is the kind of nuance that I think we should call out as a potential footgun at this phase

Reply:
They will need to bootstrap anyway because the SuperchainConfig is embedded in the OPCM. I'd rather direct everyone to bootstrap than have some folks accidentally use the wrong SuperchainConfig.

Reply:
I actually think this something we need to make more clear. There's a pretty good chance chains who wanted to maintain upgrade key ownership, just provided overrides for the L1 PAO and didn't touch the SuperchainConfig. If a chain is in that state (similar to Base and Uni's current setup), then we should make sure we don't send them down the bootstrap route.